samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -28,7 +33,101 @@ Nrf::PersistentStorageNode CreateIndexNode(uint8_t bridgedDeviceIndex, Nrf::Pers
 
 } /* namespace */
 
-namespace Nrf {
+namespace Nrf
+{
+
+bool BridgeStorageManager::Init()
+{
+	bool result = Nrf::GetPersistentStorage().NonSecureInit();
+
+	if (!result) {
+		return result;
+	}
+
+	/* Perform data migration from previous data structure versions if needed. */
+	result = MigrateData();
+
+	return result;
+}
+
+bool BridgeStorageManager::MigrateData()
+{
+	uint8_t version;
+
+	/* Check if version key is present in settings to provide backward compatibility between post-2.6.0 releases and
+	 * the previous one. If the version key is present it means that the new keys structure is used (only version
+	 * equal to 1 is for now valid). Otherwise, the deprecated structure is used, so the migration has to be done.
+	 */
+	if (!LoadDataToObject(&mVersion, version)) {
+		uint8_t count;
+		uint8_t indexes[BridgeManager::kMaxBridgedDevices] = { 0 };
+		size_t indexesCount = 0;
+
+		if (LoadBridgedDevicesCount(count) &&
+		    LoadBridgedDevicesIndexes(indexes, BridgeManager::kMaxBridgedDevices, indexesCount)) {
+			/* Load all devices based on the read count number. */
+			for (size_t i = 0; i < indexesCount; i++) {
+				BridgedDevice device;
+				if (!LoadBridgedDeviceEndpointId(device.mEndpointId, i)) {
+					return false;
+				}
+
+				/* Ignore an error, as node label is optional, so it may not be found. */
+				if (!LoadBridgedDeviceNodeLabel(device.mNodeLabel, sizeof(device.mNodeLabel),
+								device.mNodeLabelLength, i)) {
+					device.mNodeLabelLength = 0;
+				}
+
+				if (!LoadBridgedDeviceType(device.mDeviceType, i)) {
+					return false;
+				}
+
+#ifdef CONFIG_BRIDGED_DEVICE_BT
+				bt_addr_le_t btAddr;
+
+				if (!LoadBtAddress(btAddr, i)) {
+					return false;
+				}
+
+				/* Insert Bluetooth LE address as a part of implementation specific user data. */
+				device.mUserDataSize = sizeof(btAddr);
+				device.mUserData = reinterpret_cast<uint8_t *>(&btAddr);
+#endif
+
+				/* Remove all information described using an old scheme. */
```

Comment:
1. Isn't it better to first store the new data and than remove the old entries just in case the process is interrupted for some reason?
2. Also, in case you fail to load some old entries it may be worth continuing removing them instead of returning `false`. Overall, if load methods no longer support the old format, it may make the most sense to migrate as much as possible instead of failing the initialization on error.